src/JoinProcessor/SplitJoinJP.cpp

```diff
@@ -0,0 +1,147 @@
+//
+// Created by tony on 18/03/22.
+//
+
+#include <JoinProcessor/SplitJoinJP.h>
+using namespace INTELLI;
+void SplitJoinJP::joinS(TuplePtr ts) {
+  size_t timeNow = ts->subKey;
+  expireR(timeNow);
+  //single thread join window r and tuple s
+  TuplePtrQueueLocalS->push(ts);
```

Comment:
Is this the local storage of each joiner?

Reply:
If so, add some more comments.

Reply:
Do this tmr

Reply:
See the accordingly *.h